### PR TITLE
Set of fixes to build project on Linux

### PR DESCRIPTION
**Pull request by olegshtch**
Also enables CI to test build on Ubuntu 20.04

### CODE REVIEW

**Comment by longagofaraway** [comment]
Never tried running this on linux. I'll try it when I get back from vacation and fix my broken pc.

File: common/playAbility.cpp [hunk 1]
```diff
@@ -342,7 +342,7 @@ Resumable ServerPlayer::checkTiming() {
 
         if (playableCount > 1) {
             clearExpectedComands();
-            addExpectedCommand(CommandPlayAbility::GetDescriptor()->name());
+            addExpectedCommand(CommandPlayAbility::descriptor()->name());
```
Comment (longagofaraway) [inline common/playAbility.cpp:520]: Why should we use descriptor()? I can find documentation for GetDescriptor, but not for descriptor.

Comment (olegshtch) [reply common/playAbility.cpp:520]: Old protobuf versions don't support static `GetDescriptor`.

**Comment by longagofaraway** [comment]
Sorry, I had a lot of changes in master locally. I've just pushed them. Some changes are probably needed here.

**Comment by olegshtch** [comment]
I've fixed conflicts.

File: client/src/playAbility.cpp [hunk 2]
```diff
@@ -15,12 +15,12 @@ namespace {
 template<typename F>
 auto decodingWrapper(const std::string &buf, F &decodeFunction) {
     std::vector<uint8_t> binbuf(buf.begin(), buf.end());
-    auto it = binbuf.begin();
+    std::vector<uint8_t>::const_iterator it = binbuf.begin();
```
Comment (longagofaraway) [inline client/src/playAbility.cpp]: Will auto and cbegin work?

File: abilities/tools/abilityMaker/CMakeLists.txt [hunk 3]
```diff
@@ -8,9 +8,13 @@ find_package(Qt5 COMPONENTS Quick REQUIRED)
 
 add_executable(AbilityMaker ${abilityMakerSources})
 
-target_compile_features(AbilityMaker PRIVATE cxx_std_17)
+target_compile_features(AbilityMaker PRIVATE cxx_std_20)
 target_compile_definitions(AbilityMaker
   PRIVATE $<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:QT_QML_DEBUG>)
+if (UNIX)
+  target_compile_options(AbilityMaker PRIVATE -fcoroutines)
```
Comment (longagofaraway) [inline abilities/tools/abilityMaker/CMakeLists.txt:15]: This tool is supposed to be just a small application for helping creating abilities. It's not supposed to work with coroutines, so I doubt this is needed.

File: client/src/playAbility.cpp (hunk 2 above)
Comment (olegshtch) [reply client/src/playAbility.cpp]: Yes, it works.

File: abilities/tools/abilityMaker/CMakeLists.txt (hunk 3 above)
Comment (olegshtch) [reply abilities/tools/abilityMaker/CMakeLists.txt:15]: It doesn't build without coroutines.

**Comment by longagofaraway** [comment]
Ok, please combine 2 commits into one and I'll merge it.

**Comment by olegshtch** [comment]
Squashed it

**Comment by longagofaraway** [comment]
CI build works, thanks!

**Pull request merged**